nntrainer/tensor/manager.cpp

```diff
@@ -625,13 +625,51 @@ Manager::getWeights(const std::function<bool(const Weight *)> &condition) {
 }
 
 void Manager::flushCache() {
-  weight_pool.flushCache();
-  tensor_pool.flushCache();
+  if (!swap_lookahead) {
+    weight_pool.flushCache();
+    tensor_pool.flushCache();
+  }
 }
 
 void Manager::flushCacheExcept(unsigned int order) {
-  weight_pool.flushCacheExcept(order);
-  tensor_pool.flushCacheExcept(order);
+  auto loadAsync = [&](TensorPool &pool, unsigned int order) {
+    return pool.loadCacheExecAsync(
+      order, [&](int id, TaskExecutor::CompleteStatus status) {
+        std::scoped_lock<std::mutex> lock(completed_mutex);
+        completed[id].set_value(true);
+      });
+  };
+
+  auto waitComplete = [&](unsigned int o) {
+    auto &tasks = async_task_eos[o];
+
+    std::unique_lock<std::mutex> lock(completed_mutex);
+    auto w_fut = completed[std::get<0>(tasks)].get_future();
+    auto t_fut = completed[std::get<1>(tasks)].get_future();
+    lock.unlock();
+
+    w_fut.wait();
+    t_fut.wait();
+
+    async_task_eos.erase(o);
+  };
+
+  // TODO: lookahead > 1 is required.
```

Comment:
I agree we need to do supporting when lookahead is greater than 1. and dynamic setting the Look a Head.

Reply:
OK^^